.npmrc

```diff
@@ -0,0 +1,3 @@
+# set git commit when bump version
+message="Bump version to %s"
```

Comment:
Can this file be removed?

Reply:
Don't you like an extra config file for npm?
if you didn't set the `message`, the git commit message would fallback to version number only like `0.23.6` only when you bump versions.
If you still need a message says `bump version to blahblah`, you can set this option globally via running `npm config set <key> <value>`, or just add a `-m <message>` when you are bumping versions.

Reply:
> Don't you like an extra config file for npm?

:joy:

Sorry, I don't understand what message there is.  Running `npm version patch` seems to modify the files directly but does not `git add` or `git commit` them.  Is it supposed to do something else?  Even if I `npm version patch -m "message"` it does not commit.

Reply:
That's weird. What version are your nodejs and npm?
According to [npm's doc](https://docs.npmjs.com/cli/v7/commands/npm-version), when you run `npm version` command in a git repo (which should be a clean one), it would automatically create a commit and tag it.

Reply:
npm version is 6.14.8.  nodejs is 14.15.1.

Reply:
Yeah, I am using npm v7.x , and I can reproduce.
Let me figure out why and how to deal with it later.
```bash
$ nvm use 14.15.1
Now using node v14.15.1 (npm v6.14.8)

$ npm install -g npm@6.14.8
/home/***/.nvm/versions/node/v14.15.1/bin/npm -> /home/***/.nvm/versions/node/v14.15.1/lib/node_modules/npm/bin/npm-cli.js
/home/***/.nvm/versions/node/v14.15.1/bin/npx -> /home/***/.nvm/versions/node/v14.15.1/lib/node_modules/npm/bin/npx-cli.js
+ npm@6.14.8
updated 1 package in 5.345s

$ node -v
v14.15.1

$ npm -v
6.14.8
$ cat package.json | grep version
  "version": "0.23.6",
    "version": "node util/sync_version.cjs"

$ git log --oneline | head -3
b5935e4a chore: add script sync version number
865cf1ac chore: dont tag with npm
e5fb4b4f chore: add version in package.json

$ npm version minor
v0.24.0

> cactbot@0.24.0 version /workspaces/cactbot
> node util/sync_version.cjs



   ╭────────────────────────────────────────────────────────────────╮
   │                                                                │
   │      New major version of npm available! 6.14.8 → 7.24.0       │
   │   Changelog: https://github.com/npm/cli/releases/tag/v7.24.0   │
   │               Run npm install -g npm to update!                │
   │                                                                │
   ╰────────────────────────────────────────────────────────────────╯

$ cat package.json | grep version
  "version": "0.24.0",
    "version": "node util/sync_version.cjs"

$ git log --oneline | head -3
b5935e4a chore: add script sync version number
865cf1ac chore: dont tag with npm
e5fb4b4f chore: add version in package.json
```

Reply:
Ah, I know it.
That's because I set `git-tag-version` option as false in `.nvmrc`, and this option behaves differently in npm v6.x and v7.x:

- v6.x
> git-tag-version
> - Default: true
> - Type: Boolean
> Commit and tag the version change.
- v7.x
> git-tag-version
> - Default: true
> - Type: Boolean
> Tag the commit when using the npm version command.

So....can I just remove the `create-tag.yml` and let npm create a tag for you? Only you should remember to push tags as well.

Reply:
Yeah, removing `create-tag.yml` seems reasonable.  If I change `git-tag-version=true`, I get the commit properly.  Nice find!

However, the commit does not include the assembly files (which are unstaged).  Does `util/sync_version.cjs` need to `git add` or `git commit -a --amend -C HEAD` them?

Reply:
Done!
Yeah, `npm` suggest to run an explicit `git add .` too, so I add it.

Reply:
Thank you! I wasn't sure what the right thing to do was.  This works great.  I made one last edit and will merge this.  I changed `git add .` to `git add -u` so that it didn't add any loose untracked files in the repo, only tracked changed files.